content/en-us/scripting/security/security-tactics.md

```diff
@@ -113,6 +123,49 @@ local function isInf(n: number): boolean
 end
 ```
 
+Another common attack that exploiters may use involves sending `Library.table|tables` in place of an `Class.Instance`. Complex payloads can mimic what would be an otherwise ordinary object reference.
+
+For example, provided with an [in-experience shop](#in-experience-shop) system where item data like prices are stored in `Class.NumberValue` objects, an exploiter may circumvent all other checks by doing the following:
+
+```lua title="LocalScript in StarterPlayerScripts" highlight="5, 17, 20"
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+
+local itemDataFolder = ReplicatedStorage:WaitForChild("ItemData")
+local buyItemEvent = ReplicatedStorage:WaitForChild("BuyItemEvent")
+local payload = {
+	Name = "Ultra Blade",
+	ClassName = "Folder",
+	Parent = itemDataFolder,
+	Price = {
+		Name = "Price",
+		ClassName = "NumberValue",
+		Value = 0,  -- Negative values could also be used, resulting in giving currency rather than taking it!
+	},
+}
+
+-- Send malicious payload to the server (this will be rejected)
+print(buyItemEvent:InvokeServer(payload))  -- Outputs "false Invalid item provided"
+
+-- Send a real item to the server (this will go through!)
+print(buyItemEvent:InvokeServer(itemDatafolder["Real Blade"]))  -- Outputs "true" and remaining currency if purchase succeeds
+```
+
+```lua title="Script in ServerScriptService" highlight="7-10"
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+
+local itemDataFolder = ReplicatedStorage:WaitForChild("ItemData")
+local buyItemEvent = ReplicatedStorage:WaitForChild("BuyItemEvent")
+
+buyItemEvent.OnServerInvoke = function(player, item)
+	-- Check if the passed item isn't spoofed and is in the ItemData folder
+	if typeof(item) ~= "Instance" or not item:IsDescendantOf(itemDataFolder) then
+		return false, "Invalid item provided"
+	end
+
+	-- The server can then go on to process the purchase based on the example flow below
+end
```

Comment:
```suggestion
local function buyItem(player, item)
	-- Check if the passed item isn't spoofed and is in the ItemData folder
	if typeof(item) ~= "Instance" or not item:IsDescendantOf(itemDataFolder) then
		return false, "Invalid item provided"
	end

	-- The server can then go on to process the purchase based on the example flow below
end

-- Bind "buyItem()" to the remote function's callback
buyItemEvent.OnServerInvoke = buyItem
```

Reply:
Basically just declaring a dedicated non-anonymous function to fit style of example above.

Reply:
Very good change! Allows for usage from other potential avenues of purchasing (bulk purchases etc.)